### PR TITLE
Added support for .mjs files

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
.mjs an extension for EcmaScript modules.
It's a source code file containing an ES Module (ECMAScript Module) for use with a Node.js application.
These files are written in JavaScript, and may also use the .js extension outside the Node.js context.

We never added them beforehand, although they contain regular Javascript code.

_This may lead to more findings for customers._

### CODE REVIEW

File: src/main/scala/io/shiftleft/js2cpg/io/FileUtils.scala [hunk 1]
```diff
@@ -109,7 +108,7 @@ object FileUtils {
   )(implicit
     copyOptions: File.CopyOptions = File.CopyOptions(false)): File = {
     val fileCollector = FileCollector(
-      PathFilter(from.path, config, filterIgnoredFiles = false, None))
+      PathFilter(from.path, config, filterIgnoredFiles = false, List.empty))
```
Comment (maltek) [inline src/main/scala/io/shiftleft/js2cpg/io/FileUtils.scala]:
```suggestion
      PathFilter(from.path, config, filterIgnoredFiles = false, extensions = List.empty))
```

File: src/main/scala/io/shiftleft/js2cpg/io/PathFilter.scala [hunk 2]
```diff
@@ -39,11 +39,13 @@ case class PathFilter(rootPath: Path,
...
-    case Some(ext) =>
-      file.isRegularFile && !file.extension.contains(DTS_SUFFIX) && file.toString.endsWith(ext)
-    case None =>
+  private def acceptFile(file: File): Boolean = extensions match {
+    case Nil =>
       file.isRegularFile && !file.extension.contains(DTS_SUFFIX)
+    case exts =>
+      file.isRegularFile &&
+        !file.extension.contains(DTS_SUFFIX) &&
+        exts.exists(file.toString.endsWith)
```
Comment (maltek) [inline src/main/scala/io/shiftleft/js2cpg/io/PathFilter.scala]:
```suggestion
  private def acceptFile(file: File): Boolean = file.isRegularFile &&
        !file.extension.contains(DTS_SUFFIX) &&
        (extensions.isEmpty || file.extension.exists(extensions.contains))
```

**Review by maltek — DISMISSED** [review]
I just have some stylistic suggestions, feel free to disregard if you disagree

**Comment by max-leuthaeuser** [comment]
> I just have some stylistic suggestions, feel free to disregard if you disagree

Thanks!

**Review by maltek — APPROVED** [review]

**Pull request merged**